core/tests/test_window.py

```diff
@@ -381,6 +381,49 @@ def test_screen_position(window, app):
     assert window.screen_position == (100, 100)
 
 
+def test_widget_id_reusable_after_window_closes(window, app):
+    """Widget IDs can be reused after the associated widget's window is closed."""
+
+    # Common widget IDs
+    common_widget_ids = {"content": "window_content", "label": "sample_label"}
+
+    # Create a second window and create its content and children & specify their IDs.
+    second_window = toga.Window()
+    second_window.content = toga.Box(
+        id=common_widget_ids["content"],
+        children=[toga.Label(text="Sample Label", id=common_widget_ids["label"])],
+    )
+    # Show the second window and check that it is in the app's windows registry.
```

Comment:
There seems to be some confusion here. We aren't concerned about the *window* registry at all (not that there even is one. There's a list of windows). 

What we care about is the *widget* registry. The tests you've got here verify the *window* has been created, but that's not the thing that is the source of the bugs here.